### PR TITLE
Add an option for vue-router

### PR DESCRIPTION
**Pull request by crswll**
A lot of people ask how to set up the router in gitter.im so I figured an option here would be nice.

### CODE REVIEW

**Comment by gkatsanos** [comment]
It would be very nice, I m just starting with vue-router and there's nothing useful in the vue-router documention regarding our webpack setup.

File: template/src/router/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import Vue from 'vue'{{#if_eq lintConfig "airbnb"}};{{/if_eq}}
+import Router from 'vue-router'{{#if_eq lintConfig "airbnb"}};{{/if_eq}}
+
+Vue.use(Router){{#if_eq lintConfig "airbnb"}};{{/if_eq}}
+
+import Hello from '../components/Hello'{{#if_eq lintConfig "airbnb"}};{{/if_eq}}
+
+export default new Router({
+  mode: 'history',
```
Comment (pholly) [inline template/src/router/index.js]: Yeah probably best to leave it to Router's default

**Comment by pholly** [comment]
@gkatsanos I added vue-router by following its documentation and didn't have any problems. Other than adding vue-router to `package.json` by running `npm install --save-dev vue-router` I didn't change any webpack setup.

File: template/src/router/index.js (hunk 1 above)
Comment (crswll) [inline template/src/router/index.js]: Agreed. I'll remove it.

**Comment by gkatsanos** [comment]
seems like this was merged after all ?..

**Comment by pksunkara** [comment]
Heh. Looks like I was too early with my vue-router PR. https://github.com/vuejs-templates/webpack/pull/296

Seriously don't understand why it wasn't accepted then but is accepted now.

**Comment by yyx990803** [comment]
@pksunkara apologies that I didn't pay more attention to your PR earlier - it does look very similar to this one. That said, it by no means indicate that your contribution was not welcome - it's mostly that my thoughts on what the template should include do change over time. At the time of your PR the 2.x router wasn't considered very stable but now it is - so I'm more confident in including as part of the default options now than I was a few months ago. Hope that explains it.

**Comment by pksunkara** [comment]
@yyx990803 No problem. If you can, please read through [this](https://github.com/vuejs-templates/webpack/pull/296#discussion_r82073211) comment on that PR and tell me what your thoughts are about it. Thanks.

**Pull request merged**